src/demo/index.js

```diff
@@ -1,9 +1,14 @@
 import 'whatwg-fetch';
-import { renderApp } from '../app/app';
+import ReactDOM from 'react-dom';
+import { createApp } from '../app';
+
+function renderComponent(react, id) {
+  ReactDOM.render(react, document.getElementById(id));
+}
 
 const urlParams = new URLSearchParams(window.location.search);
 if (urlParams.has('small')) {
-  renderApp('small-app', 100);
+  renderComponent(createApp(100), 'small-app');
 } else {
-  renderApp('full-app');
+  renderComponent(createApp(), 'full-app');
 }
```

Comment:
I liked having this functionality bundled as it made the non-react use-case (embedding on a web-page simpler).  Maybe `validateAndReturn` should be a simple component (so basically a name-change) but I would be curious to hear others' take.  This is just semantics so definitely not blocking your merge.